src/circulating_supply/circulating_supply_provider.rs

```diff
@@ -0,0 +1,150 @@
+use bigdecimal::{BigDecimal, ToPrimitive};
+use diesel::PgConnection;
+
+use crate::circulating_supply::lockup::{
+    get_account_state, get_code_version, is_bug_inside, TRANSFERS_ENABLED,
+};
+use crate::circulating_supply::user_balance::get_user_balance;
+use crate::db_adapters::accounts::collect_lockups_for_block;
+use crate::db_adapters::blocks::closest_block_for;
+use crate::db_adapters::circulating_supply::{
+    add_circulating_supply, get_precomputed_circulating_supply,
+};
+use crate::models;
+use crate::models::circulating_supply::CirculatingSupply;
+use crate::models::Block;
+use actix::Addr;
+use actix_web::rt::time;
+use std::ops::Div;
+use std::str::FromStr;
+use std::time::SystemTime;
+use tracing::info;
+
+const DAY: std::time::Duration = std::time::Duration::from_secs(86400);
+
+pub(crate) async fn compute_circulating_supply(view_client: Addr<near_client::ViewClientActor>) {
+    let mut current_day = TRANSFERS_ENABLED;
+    while current_day < get_now_timestamp() {
+        compute_circulating_supply_for(&view_client, current_day).await;
+        current_day += DAY.as_nanos() as u64;
+    }
+
+    // We ping the number twice a day to be sure that we computed it, and to deliver it earlier.
+    // If the value is already in the DB, we will just return almost immediately
+    let mut interval = time::interval(DAY.div(2));
```

Comment:
I suggest we introduce a separate const for the interval instead of crafting it ad-hoc.

Also, if we restart the indexer at 11:59pm, we will only get the circulating supply for that day 12 hours later, though we could do that in 2 minutes from that time, so I suggest we have a waiting loop that would check the finalized block timestamp and let it go once we cross the day boundary.